stock_production_lot_warranty/models/stock_production_lot.py

```diff
@@ -15,34 +15,25 @@ class StockProductionLot(models.Model):
 
     warranty_exp_date = fields.Date(string="Warranty Expiration Date")
 
+    def _get_warranty_exp_date(self, start_date):
+        if not self.product_id:
+            return False
+        duration_type = self.product_id.product_tmpl_id.warranty_type
+        duration = self.product_id.product_tmpl_id.warranty
+        if not duration_type or not duration:
+            return False
+        if duration_type == "day":
+            warranty_delta = timedelta(days=duration)
+        elif duration_type == "week":
+            warranty_delta = timedelta(weeks=duration)
```

Comment:
Wouldn't it make more sense to harmonize delta types and use `relativedelta` instead of `timedelta`?